usaspending_api/recipient/delta_models/recipient_lookup.py

```diff
@@ -1,30 +1,583 @@
+            ROW_NUMBER() OVER (PARTITION BY uei, awardee_or_recipient_uniqu ORDER BY update_date DESC NULLS LAST) AS row_num
+        FROM raw.sam_recipient
+        WHERE COALESCE(uei, awardee_or_recipient_uniqu) IS NOT NULL AND legal_business_name IS NULL
+    ),
+    latest_tx_no_name AS (
+        SELECT
+            6 AS priority,
+            recipient_hash,
+            duns_recipient_hash,
+            awardee_or_recipient_legal AS legal_business_name,
+            awardee_or_recipient_uniqu AS duns,
+            uei,
+            source,
+            address_line_1,
+            address_line_2,
+            NULL AS business_types_codes,  -- transactions only have business_categories, not the type codes
+            city,
+            congressional_district,
+            country_code,
+            ultimate_parent_unique_ide AS parent_duns,
+            ultimate_parent_legal_enti AS parent_legal_business_name,
+            ultimate_parent_uei AS parent_uei,
+            state,
+            zip4,
+            zip5,
+            action_date AS update_date,
+            NULL AS alternate_names,
+            ROW_NUMBER() OVER (PARTITION BY recipient_hash ORDER BY action_date DESC, is_fpds, transaction_unique_id) AS row_num
+        FROM temporary_transaction_recipients_view
+        WHERE COALESCE(uei, awardee_or_recipient_uniqu) IS NOT NULL AND awardee_or_recipient_legal IS NULL
+    ),
+    latest_duns_sam_parent_no_name AS (
+        SELECT
+            7 AS priority,
+            REGEXP_REPLACE(
+                MD5(UPPER(
+                    CASE WHEN ultimate_parent_uei IS NOT NULL THEN CONCAT('uei-', ultimate_parent_uei)
+                    ELSE CONCAT('duns-', ultimate_parent_unique_ide) END
+                )),
+                '^(\.{{8}})(\.{{4}})(\.{{4}})(\.{{4}})(\.{{12}})$',
+                '\$1-\$2-\$3-\$4-\$5'
+            ) AS recipient_hash,
+            REGEXP_REPLACE(
+                MD5(UPPER(CONCAT('duns-', ultimate_parent_unique_ide))),
+                '^(\.{{8}})(\.{{4}})(\.{{4}})(\.{{4}})(\.{{12}})$',
+                '\$1-\$2-\$3-\$4-\$5'
+            ) AS duns_recipient_hash,
+            NULL AS legal_business_name,
+            ultimate_parent_unique_ide AS duns,
+            ultimate_parent_uei AS uei,
+            'sam-parent' AS source,
+            NULL AS address_line_1,
+            NULL AS address_line_2,
+            NULL AS business_types_codes,
+            NULL AS city,
+            NULL AS congressional_district,
+            NULL AS country_code,
+            ultimate_parent_unique_ide AS parent_duns,
+            NULL AS parent_legal_business_name,
+            ultimate_parent_uei AS parent_uei,
+            NULL AS state,
+            NULL AS zip4,
+            NULL AS zip5,
+            update_date,
+            NULL AS alternate_names,
+            ROW_NUMBER() OVER (PARTITION BY ultimate_parent_uei, ultimate_parent_unique_ide ORDER BY update_date DESC NULLS LAST) AS row_num
```

Comment:
Same thing here, since one could be null, `PARTITION BY COALESCE(ultimate_parent_uei, ultimate_parent_unique_ide) ORDER BY update_date DESC NULLS LAST) AS row_num`